tests/unit/test_charm.py

```diff
@@ -267,6 +267,54 @@ def test__on_loki_push_api_endpoint_departed(
         self.assertEqual(path, "/etc/agent/agent.yaml")
         self.assertTrue(yaml.safe_load(content)["loki"] == {})
 
+    @responses.activate
+    @patch.object(Container, "pull", new=pull_empty_fake_file)
```

Comment:
Container puss and pull is now supported by harness, see e.g. https://github.com/canonical/grafana-agent-k8s-operator/pull/17/commits/621103cb14fa759dce2ee87bca28e1274241c3ae

Reply:
We will address the removal of all these mocks in https://github.com/canonical/grafana-agent-k8s-operator/issues/45